### ISSUE TITLE
Improve simplify

### ISSUE THREAD

**Issue by richardbullin** [josdejong/mathjs#1620]
The below simplify methods could be improved, also theres some funny stuff going on with the implicit operators?

```javascript
math.simplify('5x/5x').toString(); // Returns "x * 5 / (5 x)"
math.simplify('5x/5x').toString({ implicit: 'hide'}); // Returns "x * 5 / (5 x)"

math.simplify('(5x)/(5x)').toString(); // Returns "x * 5 / (5 x)"
math.simplify('(5x)/(5x)').toString({ implicit: 'hide'}); // Returns "x * 5 / (5 x)"

math.simplify('(5*x)/(5*x)').toString(); // Returns "x * 5 / (5 * x)"
math.simplify('(5*x)/(5*x)').toString({ implicit: 'hide'}); // Returns "x * 5 / (5 x)"

```
I added the below rule to simplify a bit  more and thought I would share
```javascript
{l: "(n1*n2) / (n1*n2), r: "1"}
```

**Comment by josdejong** [comment]
Thanks for sharing, there is indeed room for improvement there.

Would the rule you propose also work for something like `2x/5x`? Just handling the exact same numerator and denominator feels quite limited to me. What do you think?

**Comment by ericman314** [comment]
I wonder why this wasn't caught by a rule such as `v/v = 1`? It's been forever since I've looked at simplify.js, perhaps a lot has changed?

**Comment by josdejong** [comment]
Yeah, I don't know. Maybe because the order of `x` and `5` are not the same for num/den. I have the feeling we have to dig a bit deeper to see why `simplify` is currently not able to simplify this equation.

**Comment by richbully** [comment]
@josdejong No it doesn't solve the issue you highlighted but I wouldn't say its limited as it takes care of other situations such as `(cos(x)sin(x))/(sin(x)cos(x))`, for the `2x/5x` case its a bit more specific as you are looking at ratios between constants. I would add `{ l: '(c1*n1) / (c2*n1)', r: "c1/c2" }` in addition to `{l: "(n1*n2) / (n1*n2), r: "1"}` to capture all these cases.

@ericman314  I'd say its not simplified by the rule `n/n` because you end up with 2 nodes on the numerator and denominator and that rule assumes one node on the top and bottom?

**Comment by ericman314** [comment]
Nodes should be matched recursively, so in that case both the numerator and denominator would match the `*` node and its children. But I just glanced through the rules and I didn't see `n/n`. simplify.js was refactored a while ago, so maybe that rule got lost or was no longer needed for some reason.

**Comment by richbully** [comment]
@ericman314 Interesting! I just added that rule in and it still doesn't solve `'5x/5x'` but does work for `(cos(x)sin(x))/(sin(x)cos(x))` the nodes on the numerator were coming out as symbol * constant while the denominator was coming out as constant * symbol and one was implicit and the other wasn't. I changed the order so the top and bottom both used same order and also tried with both implicits set to false and then tried simplifying and it still didnt work so must be something going on in simplify method. Cheers guys!
